src/posthog-core.js

```diff
@@ -404,7 +408,17 @@ PostHogLib.prototype._send_request = function (url, data, options, callback) {
         window.navigator.sendBeacon(url, encodePostData(data, { ...options, sendBeacon: true }))
     } else if (USE_XHR) {
         try {
-            xhr(url, data, this.get_config('xhr_headers'), options, this._captureMetrics, callback)
+            const retryRequestId = Math.floor(Math.random() * 1000000)
```

Comment:
Since we need unique IDs anyway, thoughts on going a `uuid` route instead of this?

Reply:
Could certainly do so - just didn't think we needed any added complexity here.

Reply:
Hmm, where I'm coming from is about preventing clashes: a lot easier to overwrite things with 1 in 10^6 vs 1 in 10^{a lot}. 

Expectation of clash right now is after 1 million requests, which isn't much?

10^9 is a good compromise I think: keeps things simple + reduces clash expectation to 1 in a billion.

Reply:
Changed to 10^9, thanks!